### PR TITLE
Fix blocking channel

### PR DESCRIPTION
**Pull request by hectorhuertas**

### CODE REVIEW

File: credentials.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ func (cr *CredentialsRenewer) Start() {
 				return
 			}
 
-			log.Printf("credentials: %v", secret.Data["access_key"].(string))
+			log.Printf("new aws credentials: %v, expiring %v", secret.Data["access_key"].(string), l.Data.ExpireTime.Format("2006-01-02 15:04:05"))
```
Comment (george-angel) [inline credentials.go]: I'm picking, but if you know its going to be a string, use `%s` )

File: credentials.go [hunk 2]
```diff
@@ -88,7 +88,9 @@ func (cr *CredentialsRenewer) Start() {
 			}
 
 			// Sleep until its time to renew the creds
-			time.Sleep(sleepDuration(leaseDuration, cr.Rand))
+			sd := sleepDuration(leaseDuration, cr.Rand)
+			log.Printf("credentials renewer sleeping for: %v", sd)
```
Comment (george-angel) [inline credentials.go]: I don't think this is needed.

File: login.go [hunk 3]
```diff
@@ -51,6 +51,6 @@ func (lr *LoginRenewer) login() (time.Duration, error) {
 
 	lr.Client.SetToken(secret.Auth.ClientToken)
 
-	log.Printf("login: lease duration: %v", secret.Auth.LeaseDuration)
+	log.Printf("new vault login: lease duration: %vs", secret.Auth.LeaseDuration)
```
Comment (george-angel) [inline login.go]: `%vs`?

Comment (hectorhuertas) [reply login.go]: `$v` for the number of seconds and `s` is the unit

File: credentials.go (hunk 2 above)
Comment (hectorhuertas) [reply credentials.go]: Ok, I'll revert this one

**Pull request merged**